Beta 1/ RC1 doesn't run on macos 10.13.6, it can't find the symbol for timespec_get()

Beta 1/ RC1doesn't run on macos 10.13.6, it can't find the symbol for timespec_get()

Alpha 3 works just fine.

Looking at the headers

#if (__DARWIN_C_LEVEL >= __DARWIN_C_FULL) || \
        (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L) || \
        (defined(__cplusplus) && __cplusplus >= 201703L)
/* ISO/IEC 9899:201x 7.27.2.5 The timespec_get function */
#define TIME_UTC        1       /* time elapsed since epoch */
__API_AVAILABLE(macosx(10.15), ios(13.0), tvos(13.0), watchos(6.0))
int timespec_get(struct timespec *ts, int base);
#endif

so it was first available on 10.15. what is the oldest version of macos that kodi is to support?

From a Kodi team member in the forums. @kambala-decapitator (I think) [https://forum.kodi.tv/showthread.php?tid=370404]


that's a bug indeed, thanks for discovering!

also see this in my local tvOS build, seems coming from libmicrohttpd. I wonder why it's a warning and not an error...
configure:26454: checking for C11 timespec_get()
configure:26486:  /Applications/Xcode13.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -o conftest -no-cpp-precomp -fno-common -fheinous-gnu-extensions -ftree-vectorize -pipe -Wno-trigraphs -fpascal-strings -Wreturn-type -Wunused-variable -fmessage-length=0 -gdwarf-2 -g -D_DEBUG  -arch arm64 -mappletvos-version-min=11.0 -isysroot /Applications/Xcode13.4.1.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS15.4.sdk -isystem /Users/Shared/xbmc-depends/appletvos15.4_arm64-target-debug/include  -fno-strict-aliasing -fheinous-gnu-extensions -ftree-vectorize -pipe -Wno-trigraphs -fpascal-strings -Wreturn-type -Wunused-variable -fmessage-length=0 -gdwarf-2 -g -D_DEBUG  -arch arm64 -mappletvos-version-min=11.0 -isysroot /Applications/Xcode13.4.1.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS15.4.sdk -isystem /Users/Shared/xbmc-depends/appletvos15.4_arm64-target-debug/include  -L/Users/Shared/xbmc-depends/appletvos15.4_arm64-target-debug/lib -Wl,-search_paths_first -L/Applications/Xcode13.4.1.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS15.4.sdk/usr/lib -arch arm64 -mappletvos-version-min=11.0 -isysroot /Applications/Xcode13.4.1.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS15.4.sdk -stdlib=libc++  -liconv   conftest.c  >&5
conftest.c:124:19: warning: 'timespec_get' is only available on tvOS 13.0 or newer [-Wunguarded-availability-new]
  if (TIME_UTC != timespec_get (&ts, TIME_UTC))
                  ^~~~~~~~~~~~
/Applications/Xcode13.4.1.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS15.4.sdk/usr/include/time.h:199:5: note: 'timespec_get' has been marked as being introduced in tvOS 13.0 here, but the deployment target is tvOS 11.0.0
int timespec_get(struct timespec *ts, int base);
    ^
conftest.c:124:19: note: enclose 'timespec_get' in a __builtin_available check to silence this warning
  if (TIME_UTC != timespec_get (&ts, TIME_UTC))
                  ^~~~~~~~~~~~
1 warning generated.
configure:26486: $? = 0
configure:26498: result: yes

--- a/tools/depends/target/libmicrohttpd/Makefile
+++ b/tools/depends/target/libmicrohttpd/Makefile
@@ -15,6 +15,13 @@ ifeq ($(DEBUG_BUILD), yes)
   CONFIGURE+= --enable-asserts
 endif
 
+ifeq ($(findstring apple-darwin, $(HOST)), apple-darwin)
+  # blanket disable timespec_get use for apple platforms. timespec_get was introduced in
+  # __API_AVAILABLE(macosx(10.15), ios(13.0), tvos(13.0), watchos(6.0)) but older platforms
+  # are failing to run.
+  CONFIGURE+= mhd_cv_func_timespec_get=no
+endif
+
 LIBDYLIB=$(PLATFORM)/src/microhttpd/.libs/$(LIBNAME).a
 
 all: .installed-$(PLATFORM)

